pkg/traceql/ast_execute.go

```diff
@@ -422,6 +422,28 @@ func (o *BinaryOperation) execute(span Span) (Static, error) {
 		}
 	}
 
+	if lhsT.isMatchingArrayElement(rhsT) {
+		var (
+			res Static
+			err error
+		)
+
+		// TODO: this can be a range loop once iterators are available in Go 1.23
+		//       for _, s := range lhs.Elements() { ... }
+		lhs.Elements()(func(_ int, elem Static) bool {
```

Comment:
> if the goal is to support only boolean operations i would prefer something more explicit
limited it to boolean operation, it will error out if someone does non-boolean operation on arrays.

Added bunch more tests and made it work symmetrically (with tests)